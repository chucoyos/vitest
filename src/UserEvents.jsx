import { useState } from 'react'
function UserEvents() {
	const [text, setText] = useState('Initial Text')
	const handleText = () => {
		setText('Modified text')
	}
	return (
		<div>
			<h1>{text}</h1>
			<button onClick={handleText}>Set Text</button>
		</div>
	)
}

export default UserEvents
