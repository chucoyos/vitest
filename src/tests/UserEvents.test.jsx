import UserEvents from '../UserEvents'
import { it, describe, expect } from 'vitest'
import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

describe('UserEvents component', () => {
	it('Renders Modified text', () => {
		const { container } = render(<UserEvents />)
		expect(container).toMatchSnapshot()
	})
	it('Renders Modified text after a button click', async () => {
		const user = userEvent.setup()
		render(<UserEvents />)
		const button = screen.getByRole('button', { name: 'Set Text' })
		await user.click(button)
		expect(screen.getByRole('heading').textContent).toMatch(/Modified Text/i)
	})
})
