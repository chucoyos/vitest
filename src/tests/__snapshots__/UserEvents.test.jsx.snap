// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`UserEvents component > Renders Modified text 1`] = `
<div>
  <div>
    <h1>
      Initial Text
    </h1>
    <button>
      Set Text
    </button>
  </div>
</div>
`;
